rust/nasl-interpreter/src/built_in_functions/packet_forgery.rs

```diff
@@ -0,0 +1,2872 @@
+    if positional.is_empty() {
+        return Ok(NaslValue::Null);
+    }
+
+    for ip in positional.iter() {
+        match ip {
+            NaslValue::Data(data) => {
+                let pkt = packet::ipv4::Ipv4Packet::new(data).ok_or_else(|| {
+                    FunctionErrorKind::Diagnostic(
+                        "No possible to create a packet from buffer".to_string(),
+                        Some(NaslValue::Null),
+                    )
+                })?;
+                println!("------\n");
+                println!("\tip_hl  : {:?}", pkt.get_header_length());
+                println!("\tip_v   : {:?}", pkt.get_version());
+                println!("\tip_tos : {:?}", pkt.get_dscp());
+                println!("\tip_len : {:?}", pkt.get_total_length());
+                println!("\tip_id  : {:?}", pkt.get_identification());
+                println!("\tip_off : {:?}", pkt.get_fragment_offset());
+                println!("\tip_ttl : {:?}", pkt.get_ttl());
+                // TODO: match pkt.get_next_level_protocol()
+                println!("\tip_sum  : {:?}", pkt.get_checksum());
+                println!("\tip_src : {:?}", pkt.get_source().to_string());
+                println!("\tip_dst : {:?}", pkt.get_destination().to_string());
+                display_packet(data);
+            }
+            _ => {
+                return Err(FunctionErrorKind::WrongArgument(
+                    "valid ip packet".to_string(),
+                ));
+            }
+        }
+    }
+
+    Ok(NaslValue::Null)
+}
+
+/// Add a option to a specified IP datagram.
+///  
+/// - ip: is the IP datagram
+/// - code: is the identifier of the option to add
+/// - length: is the length of the option data
+/// - value: is the option data
+fn insert_ip_options<K>(
+    register: &Register,
+    _configs: &Context<K>,
+) -> Result<NaslValue, FunctionErrorKind> {
+    let buf = match register.named("ip") {
+        Some(ContextType::Value(NaslValue::Data(d))) => d.clone(),
+        _ => {
+            return Err(FunctionErrorKind::Diagnostic(
+                "Usage : insert_ip_options(ip:<ip>, code:<code>, length:<len>, value:<value"
+                    .to_string(),
+                Some(NaslValue::Null),
+            ));
+        }
+    };
+
+    let code = match register.named("code") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x,
+        _ => {
+            return Err(FunctionErrorKind::Diagnostic(
+                "Usage : insert_ip_options(ip:<ip>, code:<code>, length:<len>, value:<value"
+                    .to_string(),
+                Some(NaslValue::Null),
+            ));
+        }
+    };
+    let length = match register.named("length") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as usize,
+        _ => {
+            return Err(FunctionErrorKind::Diagnostic(
+                "Usage : insert_ip_options(ip:<ip>, code:<code>, length:<len>, value:<value"
```

Comment:
It seems wrong on an usage error to not abort but continue with a default? 

I think aborting seems to be the better choice? 

Even when not I think we should introduce a special case rather than using diagnostic everywhere.